sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldNotBeIgnored.cs

```diff
@@ -44,13 +44,15 @@ public sealed class TestMethodShouldNotBeIgnored : SonarDiagnosticAnalyzer
 
         private static readonly ISet<KnownType> TrackedTestMethodAttributes = new HashSet<KnownType>
         {
+            // xUnit has it's own "ignore" mechanism (by providing a (Skip = "reason") string in
+            // the attribute, so there is always an explanation for the test being skipped).
             KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_TestMethodAttribute,
+            KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_DataTestMethodAttribute,
             KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_TestClassAttribute,
             KnownType.NUnit_Framework_TestAttribute,
             KnownType.NUnit_Framework_TestCaseAttribute,
             KnownType.NUnit_Framework_TestCaseSourceAttribute,
-            KnownType.Xunit_FactAttribute,
-            KnownType.Xunit_TheoryAttribute
+            KnownType.NUnit_Framework_TheoryAttribute

```

Comment:
Added support for two missing attributes types.
Removed xUnit attributes from the scope of the rule; it isn't possible to ignore an xUnit test without providing a reason, and applying MSTest/NUnit _Ignore_ attributes against an xUnit test does nothing.